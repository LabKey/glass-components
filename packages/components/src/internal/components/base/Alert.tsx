/*
 * Copyright (c) 2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React, { FC, HTMLProps, ReactNode } from 'react';
import classNames from 'classnames';

export interface AlertProps extends HTMLProps<HTMLDivElement> {
    bsStyle?: string;
    closeLabel?: ReactNode;
    onDismiss?: () => void;
}

/**
 * An Alert that will only display if children are available. Defaults to bsStyle "danger".
 */
export const Alert: FC<AlertProps> = props => {
    const { bsStyle, children, className, closeLabel, onDismiss, ...divProps } = props;
    if (!children) return null;

    const dismissible = !!onDismiss;
    return (
        <div
            {...divProps}
            className={classNames(className, `alert alert-${bsStyle}`, { 'alert-dismissable': dismissible })}
            role="alert"
        >
            {dismissible && (
                <button className="close" onClick={onDismiss} type="button">
                    <span aria-hidden="true">&times;</span>
                    <span className="sr-only">{closeLabel}</span>
                </button>
            )}
            {children}
        </div>
    );
};

Alert.defaultProps = {
    bsStyle: 'danger',
};

Alert.displayName = 'Alert';
