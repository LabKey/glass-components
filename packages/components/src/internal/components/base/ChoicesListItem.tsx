import React, { FC, memo, ReactNode, useCallback } from 'react';
import classNames from 'classnames';

interface Props {
    active: boolean;
    componentRight?: ReactNode;
    index: number;
    label: string;
    subLabel?: string;
    onSelect: (index: number) => void;
    disabled?: boolean;
}

export const ChoicesListItem: FC<Props> = memo(props => {
    const { label, index, active, onSelect, subLabel, componentRight, disabled } = props;
    const onClick = useCallback(() => {
        onSelect(index);
    }, [onSelect, index]);

    return (
        <button className={classNames('list-group-item', { active })} onClick={onClick} type="button" disabled={disabled}>
            {label}
            {subLabel && <span className="choices-list__sub-label">{subLabel}</span>}
            {componentRight}
        </button>
    );
});
ChoicesListItem.displayName = 'ChoicesListItem';
