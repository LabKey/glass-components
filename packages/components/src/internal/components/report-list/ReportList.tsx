/*
 * Copyright (c) 2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React, { PureComponent } from 'react';
import { Link } from 'react-router';
import { Image, Media, Modal, Panel } from 'react-bootstrap';

import { ActionURL, Ajax, Utils } from '@labkey/api';

import { PreviewGrid } from '../PreviewGrid';
import { Chart } from '../chart/Chart';

import { DataViewInfo, IDataViewInfo } from '../../DataViewInfo';
import { DataViewInfoTypes, GRID_REPORTS, VISUALIZATION_REPORTS } from '../../constants';
import { LoadingSpinner } from '../base/LoadingSpinner';
import { Alert } from '../base/Alert';
import { SVGIcon } from '../base/SVGIcon';
import { SchemaQuery } from '../../../public/SchemaQuery';

const ICONS = {
    [DataViewInfoTypes.AutomaticPlot]: 'xy_line',
    [DataViewInfoTypes.BarChart]: 'bar_chart',
    [DataViewInfoTypes.BoxAndWhiskerPlot]: 'box_plot',
    [DataViewInfoTypes.Dataset]: 'custom_grid',
    [DataViewInfoTypes.PieChart]: 'pie_chart',
    [DataViewInfoTypes.Query]: 'custom_grid',
    [DataViewInfoTypes.SampleComparison]: 'sample_comparison',
    [DataViewInfoTypes.TimeChart]: 'xy_line',
    [DataViewInfoTypes.XYScatterPlot]: 'xy_scatter',
    [DataViewInfoTypes.XYSeriesLinePlot]: 'xy_line',
};

interface ReportConsumer {
    report: IDataViewInfo;
}

interface ReportItemModalProps extends ReportConsumer {
    onClose?(): void;
}

class ReportLinks extends PureComponent<ReportConsumer> {
    render() {
        const { runUrl, appUrl } = this.props.report;
        let appLink;

        if (appUrl) {
            appLink = (
                <p>
                    <Link to={appUrl.toString()}>View in Biologics</Link>
                </p>
            );
        }

        return (
            <div className="report-item__links">
                <p>
                    <a href={runUrl}>View in LabKey Server</a>
                </p>
                {appLink}
            </div>
        );
    }
}

class ReportMetadata extends PureComponent<ReportConsumer> {
    render() {
        const { description, type, createdBy } = this.props.report;

        return (
            <div className="report-item__metadata">
                <div className="report-item__metadata-item">
                    <label>Created By:</label>
                    <span>{createdBy}</span>
                </div>

                <div className="report-item__metadata-item">
                    <label>Type:</label>
                    <span>{type}</span>
                </div>

                <div className="report-item__metadata-item">
                    <label>Description:</label>
                    <span>{description}</span>
                </div>
            </div>
        );
    }
}

class UnsupportedReportBody extends PureComponent<ReportConsumer> {
    render() {
        return (
            <div className="report-list__unsupported-preview">
                <div className="alert alert-warning unsupported-alert">
                    <div className="unsupported-alert__icon">
                        <span className="fa fa-exclamation-circle" />
                    </div>

                    <p className="unsupported-alert__message">
                        This report is not currently supported. It is recommended that you view the report in LabKey
                        Server.
                    </p>

                    <div className="unsupported-alert__view-link">
                        <a href={this.props.report.runUrl} className="btn btn-warning">
                            View in LabKey
                        </a>
                    </div>
                </div>

                <ReportMetadata report={this.props.report} />
            </div>
        );
    }
}

class GridReportBody extends PureComponent<ReportConsumer> {
    render() {
        const { schemaName, queryName, viewName, runUrl, appUrl } = this.props.report;
        const schemaQuery = SchemaQuery.create(schemaName, queryName, viewName);

        return (
            <div className="report-list__grid-preview">
                <ReportLinks report={this.props.report} />

                <PreviewGrid schemaQuery={schemaQuery} numCols={4} numRows={3} />

                <ReportMetadata report={this.props.report} />
            </div>
        );
    }
}

class ChartReportBody extends PureComponent<ReportConsumer, any> {
    render() {
        return (
            <div className="report-list__chart-preview">
                <ReportLinks report={this.props.report} />

                <Chart chart={new DataViewInfo(this.props.report)} />

                <ReportMetadata report={this.props.report} />
            </div>
        );
    }
}

interface SampleComparisonState {
    error?: string;
    loading?: boolean;
    report?: any;
}

class SampleComparisonReportBody extends PureComponent<ReportConsumer, SampleComparisonState> {
    constructor(props) {
        super(props);

        this.state = {
            loading: false,
            report: null,
            error: null,
        };
    }

    componentDidMount() {
        this.fetchReport();
    }

    getReportId = () => {
        const { report } = this.props;
        return report.reportId.replace('db:', '');
    };

    fetchReport = () => {
        const id = this.getReportId();
        this.setState({ loading: true });

        Ajax.request({
            url: ActionURL.buildURL('assayreport', 'get.api'),
            params: { id },
            method: 'GET',
            success: Utils.getCallbackWrapper(data => {
                this.setState({ loading: false, report: data.data });
            }),
            failure: Utils.getCallbackWrapper(error => {
                this.setState({ loading: false, error: 'Error loading Sample Comparison Report' });
            }),
        });
    };

    render() {
        const { runUrl, appUrl } = this.props.report;
        const { loading, report, error } = this.state;
        const container = ActionURL.getContainer();
        const id = this.getReportId();
        const editUrl = ActionURL.buildURL('assayreport', 'edit', container, { id });
        let body = <LoadingSpinner msg="Loading report definition..." />;

        if (!loading && report) {
            const columns = report.config.selectedColumns;
            const numColumns = columns.length;
            const uniqueAssays = columns.reduce((result, column) => {
                if (column.schemaName) {
                    const schema = column.schemaName.join('/');
                    result[schema] = true;
                }
                return result;
            }, {});
            const numAssays = Object.keys(uniqueAssays).length;
            body = (
                <div>
                    This report includes {numColumns} columns from {numAssays} assays.
                </div>
            );
        } else if (error) {
            body = <Alert bsStyle="danger">{this.state.error}</Alert>;
        }

        return (
            <div className="report-list__scr-preview">
                <div className="report-item__links">
                    <p>
                        <a href={runUrl}>View in LabKey Server</a>
                    </p>
                    <p>
                        <a href={editUrl}>Edit in LabKey Server</a>
                    </p>
                    <p>
                        <Link to={appUrl.toString()}>View in Biologics</Link>
                    </p>
                </div>

                <div>{body}</div>

                <ReportMetadata report={this.props.report} />
            </div>
        );
    }
}

export class ReportItemModal extends PureComponent<ReportItemModalProps> {
    render() {
        const { name, type } = this.props.report;
        const onClose = this.props.onClose;
        let BodyRenderer = UnsupportedReportBody;

        if (GRID_REPORTS.contains(type as DataViewInfoTypes)) {
            BodyRenderer = GridReportBody;
        } else if (VISUALIZATION_REPORTS.contains(type as DataViewInfoTypes)) {
            BodyRenderer = ChartReportBody;
        } else if (type === DataViewInfoTypes.SampleComparison) {
            BodyRenderer = SampleComparisonReportBody;
        }

        return (
            <div className="report-item-modal">
                <Modal show onHide={onClose}>
                    <Modal.Header closeButton>
                        <Modal.Title>{name}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <BodyRenderer report={this.props.report} />
                    </Modal.Body>
                </Modal>
            </div>
        );
    }
}

interface ReportListItemProps {
    onClick(IDataViewInfo): void;
    report: IDataViewInfo;
}

export class ReportListItem extends PureComponent<ReportListItemProps> {
    onClick = () => this.props.onClick(this.props.report);

    onLinkClicked = e => {
        // We need to stop event propagation when clicking on a link or it will also trigger the onClick handler
        e.stopPropagation();
        return true;
    };

    render() {
        const { name, icon, iconCls, createdBy, type, runUrl, appUrl } = this.props.report;
        let nameEl = (
            <a href={runUrl} onClick={this.onLinkClicked}>
                {name}
            </a>
        );

        if (appUrl) {
            nameEl = (
                <Link to={appUrl.toString()} onClick={this.onLinkClicked}>
                    {name}
                </Link>
            );
        }

        const iconSrc = ICONS[type];
        const iconClassName = 'report-list-item__icon';
        const hasCustomIcon = icon.indexOf('reports-thumbnail.view') > -1;
        let iconEl = <Image className={iconClassName} src={icon} />;

        if (iconSrc !== undefined && !hasCustomIcon) {
            iconEl = <SVGIcon className={iconClassName} height={null} iconSrc={iconSrc} />;
        } else if (iconCls) {
            iconEl = <span className={`${iconClassName} ${iconCls} fa-4x`} />;
        }

        let createdByEl;

        if (createdBy) {
            createdByEl = <p className="report-list-item__person">Created by: {createdBy}</p>;
        }

        return (
            <Media.ListItem className="report-list-item" onClick={this.onClick}>
                <Media.Left>{iconEl}</Media.Left>

                <Media.Body>
                    <Media.Heading className="report-list-item__name">{nameEl}</Media.Heading>
                    {createdByEl}
                </Media.Body>
            </Media.ListItem>
        );
    }
}

export interface ReportListProps {
    loading: boolean;
    onReportClicked(report: IDataViewInfo): void;
    reports: IDataViewInfo[];
}

export class ReportList extends PureComponent<ReportListProps> {
    render() {
        const { loading, reports, onReportClicked } = this.props;

        let body: any;

        if (loading) {
            body = (
                <div className="report-list__message">
                    <LoadingSpinner />
                </div>
            );
        } else if (reports.length === 0) {
            body = <div className="report-list__message">No reports.</div>;
        } else {
            const reportEls = reports.map((report: IDataViewInfo) => {
                return <ReportListItem key={report.runUrl} report={report} onClick={onReportClicked} />;
            });

            body = <Media.List className="report-list__list">{reportEls}</Media.List>;
        }

        return (
            <Panel>
                <div className="report-list">{body}</div>
            </Panel>
        );
    }
}
