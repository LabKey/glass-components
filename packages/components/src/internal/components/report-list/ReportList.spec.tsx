/*
 * Copyright (c) 2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { createMemoryHistory, Route, Router } from 'react-router';
import { mount } from 'enzyme';

import exampleData from '../../../test/data/example_browse_data_tree_api.json';

import { AppURL } from '../../url/AppURL';
import { flattenBrowseDataTreeResponse } from '../../query/reports';
import { LoadingSpinner } from '../base/LoadingSpinner';

import { ReportItemModal, ReportList, ReportListItem } from './ReportList';

const history = createMemoryHistory();

const noop = () => {};
const messageSelector = '.report-list__message';
const createdBySelector = '.report-list-item__person';
const urlMapper = report => {
    const { schemaName, queryName, viewName } = report;

    if (report.type === 'Sample Comparison') {
        return AppURL.create('reports', report.reportId.replace('db:', ''));
    }

    if (!queryName) {
        return null;
    }

    const parts = ['q', schemaName, queryName];

    if (viewName) {
        parts.push(viewName);
    }

    return AppURL.create(...parts);
};

describe('<ReportList />', () => {
    test('flattenBrowseDataTreeResponse works with valid data', () => {
        flattenBrowseDataTreeResponse(exampleData, urlMapper);
    });

    test('Render with no data', () => {
        const component = <ReportList loading={false} reports={[]} onReportClicked={noop} />;
        const tree = renderer.create(component);
        expect(tree).toMatchSnapshot();
        const wrapper = mount(component);
        expect(wrapper.find(LoadingSpinner)).toHaveLength(0);
        expect(wrapper.find(messageSelector).text()).toContain('No reports');
        wrapper.unmount();
    });

    test('Render loading', () => {
        const component = <ReportList loading={true} reports={[]} onReportClicked={noop} />;
        const tree = renderer.create(component);
        expect(tree).toMatchSnapshot();
        const wrapper = mount(component);
        expect(wrapper.find(LoadingSpinner)).toHaveLength(1);
        wrapper.unmount();
    });

    test('Render with data', () => {
        const reports = flattenBrowseDataTreeResponse(exampleData, urlMapper);
        const component = <ReportList loading={false} reports={reports} onReportClicked={noop} />;
        const tree = renderer.create(component);
        expect(tree).toMatchSnapshot();
        const wrapper = mount(component);
        expect(wrapper.find(LoadingSpinner)).toHaveLength(0);
        expect(wrapper.find(ReportListItem)).toHaveLength(reports.length);
        wrapper.unmount();
    });

    test('onReportClicked should execute on click', () => {
        const reports = flattenBrowseDataTreeResponse(exampleData, urlMapper).slice(0, 1);
        const onReportClicked = jest.fn();
        const component = <ReportList loading={false} reports={reports} onReportClicked={onReportClicked} />;
        const wrapper = mount(component);
        wrapper.find(ReportListItem).simulate('click');
        expect(onReportClicked).toHaveBeenCalledTimes(1);
        wrapper.unmount();
    });
});

describe('<ReportListItem />', () => {
    test('ReportListItem renders', () => {
        const report = flattenBrowseDataTreeResponse(exampleData, urlMapper)[1];
        const onClick = jest.fn();
        const component = (
            <Router history={history}>
                <Route path="/" component={() => <ReportListItem report={report} onClick={onClick} />} />
            </Router>
        );
        const tree = renderer.create(component);
        expect(tree).toMatchSnapshot();
        const wrapper = mount(component);
        expect(wrapper.find(createdBySelector)).toHaveLength(1);
        expect(wrapper.text()).toContain(report.createdBy);
        expect(wrapper.text()).toContain(report.name);
        // Enzyme prefixes relative URLs with http://localhost
        const expectedHref = `http://localhost${report.appUrl.toString()}`;
        expect(wrapper.find('a').getDOMNode()).toHaveProperty('href', expectedHref);
        wrapper.unmount();
    });

    test('ReportListItem does not render non-existent createdBy', () => {
        const reports = flattenBrowseDataTreeResponse(exampleData, urlMapper);
        const report = reports[1];
        report.createdBy = undefined;
        const component = <ReportListItem report={report} onClick={noop} />;
        const wrapper = mount(component);
        expect(wrapper.find(createdBySelector)).toHaveLength(0);
        wrapper.unmount();
    });

    test('ReportListItem calls onClick when clicked', () => {
        const report = flattenBrowseDataTreeResponse(exampleData, urlMapper)[0];
        const onClick = jest.fn();
        const component = <ReportListItem report={report} onClick={onClick} />;
        const wrapper = mount(component);
        wrapper.simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
        // Test that we pass the report to onClick. If this test fails that means we'll need to fix any callbacks that
        // expect a report to be passed to the callback.
        expect(onClick.mock.calls[0][0]).toBe(report);
        wrapper.unmount();
    });
});

describe('<ReportItemModal />', () => {
    test('ReportItemModal renders', () => {
        // Arrange
        const report = flattenBrowseDataTreeResponse(exampleData, urlMapper)[0];
        const onClose = jest.fn();

        // Act
        const wrapper = mount(<ReportItemModal report={report} onClose={onClose} />);

        // Assert
        // Verify modal displays properly (in this case a SampleComparisonReportBody)
        expect(wrapper.find('.report-item-modal .modal-title').text()).toEqual(report.name);
        expect(wrapper.find('.report-item-modal .report-list__scr-preview').exists()).toEqual(true);

        // Verify report items are listed
        const reportItems = wrapper.find('.report-item-modal .report-item__metadata-item span');
        expect(reportItems.length).toEqual(3);
        expect(reportItems.at(0).text()).toEqual(report.createdBy);
        expect(reportItems.at(1).text()).toEqual(report.type);
        expect(reportItems.at(2).text()).toEqual(report.description);
    });
});
