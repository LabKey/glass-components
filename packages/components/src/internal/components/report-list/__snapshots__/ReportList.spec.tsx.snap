// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportList /> Render loading 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="report-list"
  >
    <div
      className="report-list__message"
    >
      <span
        className=""
      >
        <i
          aria-hidden="true"
          className="fa fa-spinner fa-pulse"
        />
         
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`<ReportList /> Render with data 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="report-list"
  >
    <ul
      className="report-list__list media-list"
    >
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <span
            className="report-list-item__icon fa fa-table fa-4x"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              href="/labkey/Biologics/assayreport-view.view?id=37"
              onClick={[Function]}
            >
              Example SCR
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Injection Volume Comparison 2
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Avg Mass vs pl
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="box_plot-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/box_plot.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Arg Box Plot
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="bar_chart-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/bar_chart.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Bar plot of Mean Cys for Runs
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="pie_chart-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/pie_chart.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Pie chart example
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="bar_chart-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/bar_chart.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Sum of Molecular Weight by Chemical Formula
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_line-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_line.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Line plot: Pro by Sample Date
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="box_plot-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/box_plot.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              MAb Box Plot
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="pie_chart-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/pie_chart.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Distribution by Lot
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Scatter: Arg vs Ala (filtered)
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Scatter: Arg vs Ala
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Injection Volume Comparison
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="box_plot-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/box_plot.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Private Chart
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Test Custom View
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Test Custom View
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Name only
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Private view
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Scientific Name only
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Name only
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              SampleID only
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={{}}
            >
              Test Custom View
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<ReportList /> Render with no data 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="report-list"
  >
    <div
      className="report-list__message"
    >
      No reports.
    </div>
  </div>
</div>
`;

exports[`<ReportListItem /> ReportListItem renders 1`] = `
<li
  className="report-list-item media"
  onClick={[Function]}
>
  <div
    className="media-left"
  >
    <img
      alt="xy_scatter-icon"
      className="report-list-item__icon"
      height={null}
      src="/labkey/_images/xy_scatter.svg"
      width="100%"
    />
  </div>
  <div
    className="media-body"
  >
    <h4
      className="report-list-item__name media-heading"
    >
      <a
        href="/q/assay.General.SEC/Data"
        onClick={[Function]}
        style={{}}
      >
        Injection Volume Comparison 2
      </a>
    </h4>
    <p
      className="report-list-item__person"
    >
      Created by: 
      alanv
    </p>
  </div>
</li>
`;
